File: ToDo App/src/Project 4/Controllers/RegisterController.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.AspNet.Identity;
using Microsoft.AspNet.Mvc;
using Project_4.Models;
using Microsoft.Data.Entity;

// For more information on enabling Web API for empty projects, visit http://go.microsoft.com/fwlink/?LinkID=397860

namespace Project_4.Controllers
{
    [Route("api/[controller]")]
    public class RegisterController : Controller
    {
        private UserManager<ToDoUser> _userManager;
        public RegisterController(UserManager<ToDoUser> userManager)
        {
            _userManager = userManager;
        }

        // POST api/register/userName/email/password
        [HttpPost("{userName}/{email}/{password}")]
        public async Task<int> Post(string userName, string email, string password)
        {
            if (await _userManager.FindByNameAsync(userName) == null)
            {
                if (await _userManager.FindByEmailAsync(email) == null) 
                {
                    var newUser = new ToDoUser()
                    {
                        UserName = userName,
                        Email = email
                    };
                    IdentityResult x = await _userManager.CreateAsync(newUser, password);
                    return 0;
                }
                return 2;
            }
            return 1;
        }
        
    }
}

File: ToDo App/src/Project 4/wwwroot/RegisterState/registerController.js
(function () {
    'use strict';

    angular
        .module('app')
        .controller('registerController', function registerController($scope, $http, $state) {

            $scope.invalidName = false;
            $scope.invalidEmail = false;

            $scope.register = function (userName, email, password) {
                $scope.invalidName = false;
                $scope.invalidEmail = false;
                $http({
                    method: 'POST',
                    url: 'api/register/' + userName + '/' + email + '/' + password
                }).success(function (data) {
                    console.log(data);
                    if (data == 1) $scope.invalidName = true;
                    else if (data == 2) $scope.invalidEmail = true;
                    else $state.go('login');
                }).error(function () {
                    console.log('failed');
                });
            }

        });
})();
